community/kernel/src/main/java/org/neo4j/kernel/impl/store/DumpStore.java

```diff
@@ -291,64 +292,45 @@ private boolean dumpRecord( STORE store, int size, StoreChannel fileChannel, Byt
             out.print( record );
             buffer.clear();
             fileChannel.read( buffer, id * size );
-            buffer.flip();
-            if ( record.inUse() )
-            {
-                dumpHex( buffer, id * size );
-            }
-            else if ( allZero( buffer ) )
-            {
-                out.printf( ": all zeros @ 0x%x - 0x%x%n", id * size, (id + 1) * size );
-            }
-            else
-            {
-                dumpHex( buffer, id * size );
-            }
+            dumpHex( record, buffer, id, size );
         }
 
         return record.inUse();
     }
 
-    private boolean allZero( ByteBuffer buffer )
+    void dumpHex( RECORD record, ByteBuffer buffer, long id, int size )
     {
-        int pos = buffer.position();
-        try
+        HexPrinter printer = new HexPrinter( out ).withBytesGroupingFormat( 16, 4, "  " ).withLineNumberDigits( 8 )
+                .withLineNumberOffset( id * size );
```

Comment:
Do we really need to instantiate and configure a new `HexPrinter` for each record?